frontend/src/components/pages/volunteer/posting/VolunteerPostingsPage.tsx

```diff
@@ -1,11 +1,240 @@
+import PostingCard from "../../../volunteer/PostingCard";
+import { PostingResponseDTO } from "../../../../types/api/PostingTypes";
+import EmptyPostingCard from "../../../volunteer/EmptyPostingCard";
+
+// delete before merging to main
+// const postingsArr = [
+//   {
+//     id: "1",
+//     skills: [
+//       { id: "1", name: "CPR" },
+//       { id: "2", name: "WHMIS" },
+//       { id: "3", name: "Photocopying" },
+//     ],
+//     branch: {
+//       id: "1",
+//       name: "branch name",
+//     },
+//     title: "Medical Reception Volunteer",
+//     startDate: new Date("Feb 28, 2022"),
+//     endDate: new Date("March 20, 2022"),
+//     autoClosingDate: new Date("Monday, November 30"),
+//     isSignedUp: false,
+//     description:
+//       "Volunteers will be responsible for updating the inventory monthly, updating the manuals and guidelines on an ongoing basis, uploading the COVID-19 screenings on a weekly basis.",
+//   },
+//   {
+//     id: "2",
+//     skills: [
+//       { id: "1", name: "CPR" },
+//       { id: "2", name: "WHMIS" },
+//       { id: "3", name: "Photocopying" },
+//     ],
+//     branch: {
+//       id: "1",
+//       name: "sample text",
+//     },
+//     title: "Medical Reception Volunteer",
+//     startDate: new Date("February 14, 2022"),
+//     endDate: new Date("March 20, 2022"),
+//     autoClosingDate: new Date("Monday, November 30"),
+//     isSignedUp: false,
+//     description:
+//       "Volunteers will be responsible for updating the inventory monthly, updating the manuals and guidelines on an ongoing basis, uploading the COVID-19 screenings on a weekly basis.",
+//   },
+//   {
+//     id: "3",
+//     skills: [
+//       { id: "1", name: "CPR" },
+//       { id: "2", name: "WHMIS" },
+//       { id: "3", name: "Photocopying" },
+//     ],
+//     branch: {
+//       id: "1",
+//       name: "sample text",
+//     },
+//     title: "Medical Reception Volunteer",
+//     startDate: new Date("March 18, 2022"),
+//     endDate: new Date("March 20, 2022"),
+//     autoClosingDate: new Date("Monday, November 30"),
+//     isSignedUp: false,
+//     description:
+//       "Volunteers will be responsible for updating the inventory monthly, updating the manuals and guidelines on an ongoing basis, uploading the COVID-19 screenings on a weekly basis.",
+//   },
+// ];
+
+// can refactor this query in the future to be more specific (depending on the data this page needs)
+const POSTINGS = gql`
+  query VolunteerPostingsPage_postings {
+    postings {
+      id
+      branch {
+        id
+        name
+      }
+      shifts {
+        id
+        postingId
+        startTime
+        endTime
+      }
+      skills {
+        id
+        name
+      }
+      employees {
+        id
+        branchId
+      }
+      title
+      type
+      status
+      description
+      startDate
+      endDate
+      autoClosingDate
+      numVolunteers
+    }
+  }
+`;
+
+const MS_PER_WEEK = 604800000;
```

Comment:
To quickly prove the correctness of calculated values, a common practice is to show the arithmetic

```suggestion
const MS_PER_WEEK = 7 * 24 * 60 * 60 * 1000;
```